clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -804,6 +804,11 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
     }
   }
 
+  if (TI.getTriple().getVendor() == Triple::AMD)
+    Builder.defineMacro("__AMD__");
+  if (TI.getTriple().getVendor() == Triple::Apple)
+    Builder.defineMacro("__APPLE__");
```

Comment:
I think this is a behavior change for non-darwin triples, but only in a "tree falls in the forest" kind of way.

@fpetrogalli you were talking about this a while back. I don't remember the verdict.

Reply:
Yes, and something that I think that is desirable in the context of Swift as well (where we are looking at some of the embedded system usage where the target triple may be something like `arm64-apple-none-macho`).

Reply:
👍